feat(aws): AWS Found Sec Best Practices & PCI DSS v3.2.1 upgrade

h4r5h1t opened the pull request:
Updated AWS Foundation Security Best Practices and PCI DSS v3.2.1 to correct control mappings

### Context

This pull request addresses the need for accurate and updated control mappings between AWS Foundation Security Best Practices and PCI DSS v3.2.1. Ensuring these mappings are correct is crucial for maintaining compliance and security posture.

### Description

This PR updates the AWS Foundation Security Best Practices and PCI DSS v3.2.1 to refine the control mappings. The previous mappings were not entirely accurate, leading to potential misinterpretations of compliance status. This change ensures that the mappings reflect the latest best practices and standards.

Dependencies: None.

### Checklist

- Are there new checks included in this PR? - No
    - If so, do we need to update permissions for the provider? Please review this carefully. - NA
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings - NA
- [ ] Review if backport is needed.
- [x] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md) - NA

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

h4r5h1t committed:
Updated AWS Foundation Security Best Practices and PCI DSS v3.2.1 to correct control mappings.

pedrooot reviewed (changes requested):
This is a great work! Thank you so much. Just two things:
1. Why are you using `"Id": "requirement_8_1"` instead of `"Id": "8.1"` we can know that this is already a requirement and this is how it's done for others compliances too.
2. Could you delete the `\u2014` that appear multiple times along the json?

Thanks for this!

h4r5h1t committed:
Merge branch 'prowler-cloud:master' into master

h4r5h1t committed:
Merge branch 'prowler-cloud:master' into master

h4r5h1t committed:
Fix compliance JSON formatting: update ID field and remove Unicode dashes

- Changed "Id": "requirement_8_1" to "Id": "8.1" to align with the standard format.
- Removed all instances of \u2014 from the JSON to eliminate unwanted characters.

h4r5h1t committed:
Update PCI 3.2.1 compliance framework description

h4r5h1t commented:
> This is a great work! Thank you so much. Just two things:
> 
> 1. Why are you using `"Id": "requirement_8_1"` instead of `"Id": "8.1"` we can know that this is already a requirement and this is how it's done for others compliances too.
> 2. Could you delete the `\u2014` that appear multiple times along the json?
> 
> Thanks for this!


Hi, thanks for the feedback! I've updated the JSON to change "Id": "requirement_8_1" to "Id": "8.1" and removed all instances of the \u2014 characters. 
Please let me know if there's anything else that needs adjustment. 

Thanks again! 🤗

pedrooot committed:
feat(dashboard): make improvements for latest changes

pedrooot reviewed (approved):
I've added some changes in order to improve Prowler Dashboard in this PR.

Thanks for your help improving compliance frameworks! 😄
